### PR TITLE
Add config option to allow upgrading legacy contracts

### PR DESCRIPTION
**Pull request by dsainati1**
Part of https://github.com/onflow/cadence/issues/2947

This adds a server config option to the emulator that allows it to parse old Cadence contracts when set, for the purposes of allowing users to test contract updates to Cadence 1.0

______

For contributor use:

- [X] Targeted PR against `master` branch
- [X] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [X] Updated relevant documentation
- [X] Re-reviewed `Files changed` in the GitHub PR explorer
- [X] Added appropriate labels

### CODE REVIEW

**Comment by dsainati1** [comment]
I am at a loss for how to do a good end to end test for this; I don't have a good way to generate old state to use as input for the test.

**Review by sideninja — APPROVED** [review]

**Comment by sideninja** [comment]
Please also add this flag to the readme

File: README.md [hunk 1]
```diff
@@ -73,6 +73,8 @@ and if you plan to run the emulator with Docker you must use the environment var
 | `--start-block-height`        | `FLOW_STARTBLOCKHEIGHT`             | `0`             | Start block height to use when starting the network using 'testnet' or 'mainnet' as the chain-id    |
 | `--evm-enabled` | `FLOW_EVMENABLED` | `false`         | Enable evm support |
 
+| `--legacy-upgrade` | `FLOW_LEGACYUPGRADE` | `false`         | Enable upgrading of legacy contracts |
```
Comment (dsainati1) [inline README.md:76]: Do I need to do anything to get the docker flag here to work?

**Review by SupunS — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0
 	github.com/improbable-eng/grpc-web v0.15.0
 	github.com/logrusorgru/aurora v2.0.3+incompatible
-	github.com/onflow/cadence v1.0.0-M3
+	github.com/onflow/cadence v1.0.0-preview.2.0.20240124204828-3adb0330b148
```
[the remark this answers is not in the document]
Comment (dsainati1) [reply go.mod]: No, I need https://github.com/onflow/cadence/pull/3043 for this. However, I think M4 was just released so I should be able to use that instead.

[the remark this answers is not in the document]
Comment (turbolent) [reply go.mod]: Yep, [M4](https://github.com/onflow/cadence/releases/v1.0.0-M4) includes that PR

File: README.md (hunk 1 above)
Comment (turbolent) [reply README.md:76]: Sorry, not quite sure what you mean here, how is this related to Docker?

**Review by turbolent — APPROVED** [review]
Nice!

**Pull request merged**